platform-services.md

```diff
@@ -0,0 +1,215 @@
+The manifest file lists the Android activities that perform the services calls and any permissions
+that your app needs to access the services.
+
+* `android/local.properties`: Contains the configuration settings to build your Android app. See
+the `hello_services` [README](https://github.com/flutter/flutter/blob/master/examples/hello_services/README.md)
+to learn more about the properties you can configure.
+
+* [`ios/HelloServices/`](https://github.com/flutter/flutter/tree/master/examples/hello_services/ios/HelloServices):
+This folder contains the Objective-C source files for the iOS-specific part of your app. The
+[`AppDelegate.m`](https://github.com/flutter/flutter/blob/master/examples/hello_services/ios/HelloServices/AppDelegate.m)
+file handles message passing between iOS and the Flutter app, while the [`LocationProvider.m`](https://github.com/flutter/flutter/blob/master/examples/hello_services/ios/HelloServices/LocationProvider.m)
+file implements the code that queries the iOS system location API
+([`CLLocationManager`](https://developer.apple.com/library/ios/documentation/CoreLocation/Reference/CLLocationManager_Class/)).
+
+* `ios/Flutter/Generated.xcconfig`: Contains the configuration settings to build your iOS app. See
+the `hello_services` [README](https://github.com/flutter/flutter/blob/master/examples/hello_services/README.md)
+to learn more about the properties you can configure.
+
+* [`lib/main.dart`](https://github.com/flutter/flutter/blob/master/examples/hello_services/lib/main.dart):
+The main entry point for your Flutter app. This source file is written in Dart syntax. The main
+function uses the [`HostMessages`](http://docs.flutter.io/flutter/services/HostMessages-class.html)class,
+which provides a high-level abstraction for passing messages to “hosts” of platform-specific
+services.
+
+## Add a service to your app {#add-a-service-to-your-app}
+
+In your Dart code (`lib/main.dart` file):
+
+1. Define a method to send a request to your target service (e.g. `_getLocation()`). You must encode
+the request as a JSON message. You’ll typically invoke this method in a
+[`State`](http://docs.flutter.io/flutter/material/State-class.html) object, when the user performs
+an operation on the UI.
+2. Define a messageHandler to process the messages returned by the target service
+(e.g. `handleGetRandom()`), then add this handler to the
+[`HostMessages`](http://docs.flutter.io/flutter/services/HostMessages-class.html) object.
+
+In your iOS code:
+
+1. In Xcode, open the `HelloServices` iOS project.
+2. Implement a service provider (e.g. `LocationProvider`). The provider class must conform to the
+`FlutterMessageListener` protocol (specifically, you must implement the
+`(NSString*)didReceiveString:(NSString*)` function). If your app wants to connect asynchronously to
+the service, use the [`FlutterAsyncMessageListener`](https://github.com/flutter/engine/blob/master/sky/shell/platform/ios/framework/Headers/FlutterAsyncMessageListener.h)
```

Comment:
First you say that it must conform to FlutterMessageListener but then you say that FlutterAsyncMessageListener is also an option.  That's contradictory.

Also, it's unclear what "your app" means in this context.  Do you mean the cross-platform part of your app?  In that case, the sentence is incorrect because the cross-platform part of your app (i.e., the part written in Dart that implements the UI) can't tell whether the platform-specific part of your app uses FlutterMessageListener or FlutterAsyncMessageListener.  In both cases, the interaction between the two is asynchronous.

Instead, it's you should use FlutterMessageListener if (1) the message does not require a reply or (2) you can reply to the message synchronously.  FlutterAsyncMessageListener is needed only when (1) the message requires a reply and (2) the code that creates the reply is asynchronous.